relax size check in flatten_for_scatter_gather

ngimel opened the pull request:
Summary: Per title, to workaround apex sbn bug.

Test Plan: Covered by existing tests

Differential Revision: D22236942

mcarilli commented on torch/lib/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -624,14 +624,10 @@ std::vector<at::Tensor> flatten_for_scatter_gather(
-      if (t.sizes() != other[i].sizes()) {
+      if (t.numel() != other[i].numel()) {
         throw std::runtime_error(
             "All tensor operands to scatter/gather must have the same size");
```
Suggestion, which you can take or leave: change "the same" to "identical"...not because it's better phrasing, but so that when people ping us with an error message, we know immediately they're using this version of the check.

Commit:
relax size check in flatten_for_scatter_gather (#40573)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/40573

Per title, to workaround apex sbn bug.

Test Plan: Covered by existing tests

Reviewed By: blefaudeux

Differential Revision: D22236942

fbshipit-source-id: 4e195e35e3e1b309400be53614d5368f6c08801e
